src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -133,8 +133,15 @@ internal void BeginDragSelection(Point2D mouseCursor)
             foreach (ISelectable selectable in DynamoSelection.Instance.Selection)
             {
                 ILocatable locatable = selectable as ILocatable;
-                if (null != locatable)
-                    draggedNodes.Add(new DraggedNode(locatable, mouseCursor));
+                if (null == locatable)
+                    continue;
+
+                // Annotations always update their position relative to all nested Nodes
+                // So there is no need to move the Annotation since it will be updated later anyway (performance improvement)
+                if (locatable is AnnotationModel)
```

Comment:
does this actually end up correctly setting the x and y of the annotationModel or does it draw it correctly, but leave the model data incorrect?

Reply:
This is how the notifications flow:
NestedNode.PositionUpdate => AnnotationNode.ModelDataUpdate => AnnotationViewModel.ViewModelDataUpdate
X and Y (and all other annotation model data) are set correctly in the annotation model then it draws (correctly).